base/iterator.jl

```diff
@@ -2,6 +2,16 @@
 
 isempty(itr) = done(itr, start(itr))
 
+_min_length(a, b, ::IsInfinite, ::IsInfinite) = min(length(a),length(b)) # inherit behaviour, error
+_min_length(a, b, A, ::IsInfinite) = length(a)
+_min_length(a, b, ::IsInfinite, B) = length(b)
+_min_length(a, b, A, B) = min(length(a),length(b))
+
+_diff_length(a, b, A, ::IsInfinite) = 0
```

Comment:
It is actually a bit tricky. One can say that  ``_diff_length(a, b, A, ::IsInfinite) = 0`` formalizes the notion that there is nothing to "collect" after the 1st, 2nd, 3rd, ..., ``n``th, ... element (i.e. all elements) from an infinite iterator are discarded or used up in some way.